Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2253,55 +2191,77 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
                                  nullptr);
     }
 
-    ComPtr<ID2D1Image> currentImage{ image };
+    auto cleanup = wil::ScopeExit([this, layer]() {
+        if (layer) {
+            this->deviceContext->PopLayer();
+        }
 
-    ComPtr<ID2D1Effect> shadowEffect;
-    RETURN_IF_FAILED(_CreateShadowEffect(currentImage.Get(), &shadowEffect));
-    if (shadowEffect) {
-        RETURN_IF_FAILED(shadowEffect.As(&currentImage));
-    }
+        // TODO GH#1194: We will need to re-evaluate Direct2D's D2DERR_RECREATE when we move to HW acceleration.
+        this->PopEndDraw();
+    });
+
+    if (state.HasShadow()) {
+        // Temporarily change the target to a command list, so that it can be 'replayed' into a shadow
+        ComPtr<ID2D1Image> originalTarget; // Cache the original target to restore it later.
+        deviceContext->GetTarget(&originalTarget);
+
+        ComPtr<ID2D1CommandList> commandList;
+        RETURN_IF_FAILED(deviceContext->CreateCommandList(&commandList));
+
+        deviceContext->SetTarget(commandList.Get());
+
+        deviceContext->SetTransform(__CGAffineTransformToD2D_F(transform));
+        RETURN_IF_FAILED(std::forward<Lambda>(drawLambda)(this, deviceContext.Get()));
```

Comment:
On failure, the transform is not restored, here and below.  I am not sure how relevant that is.